H2MLauncher.Core/ViewModels/ServerBrowserViewModel.cs

```diff
@@ -195,24 +274,30 @@ private async Task CheckUpdateStatusAsync()
         private async Task LoadServersAsync()
         {
             await _loadCancellation.CancelAsync();
+            
             _loadCancellation = new();
 
             try
             {
                 Servers.Clear();
-                TotalServers = 0;
+                TotalPlayersOverAll = 0;
+                TotalServersOverAll = 0;
                 TotalPlayers = 0;
-
+                TotalServers = 0;
                 // Get servers from the master
                 List<RaidMaxServer> servers = await _raidMaxService.GetServerInfosAsync(_loadCancellation.Token);
 
+                List<UserFavorite> userFavorites = UserFavoritesUtility.GetFavorites();
+
                 // Let's prioritize populated servers first for getting game server info.
                 IEnumerable<RaidMaxServer> serversOrderedByOccupation = servers
                     .OrderByDescending((server) => server.ClientNum);
 
                 // Start by sending info requests to the game servers
                 await _gameServerCommunicationService.StartRetrievingGameServerInfo(serversOrderedByOccupation, (server, gameServer) =>
                 {
+
+                    bool isFavorite = userFavorites.Any(fav => fav.ServerName == server.HostName);
```

Comment:
There could be servers with the same host name, also check ip and port. Or better yet, only ip and port. As the hostname could change!